### PR TITLE
use async storage api for rpc and chunk pool

### PR DESCRIPTION
**Pull request by qhz-ty**

### CODE REVIEW

File: node/chunk_pool/src/handler.rs [hunk 1]
```diff
@@ -2,22 +2,22 @@ use super::mem_pool::MemoryChunkPool;
...
+use storage_async::Store;
 use tokio::sync::mpsc::UnboundedReceiver;
 
 /// Handle the cached file when uploaded completely and verified from blockchain.
 /// Generally, the file will be persisted into log store.
 pub struct ChunkPoolHandler {
     receiver: UnboundedReceiver<DataRoot>,
     mem_pool: Arc<MemoryChunkPool>,
-    log_store: Arc<dyn Store>,
+    log_store: Arc<Store>,
```
Comment (ionian-dev) [inline node/chunk_pool/src/handler.rs:13]: No need to use `Arc` here.

File: node/chunk_pool/src/handler.rs [hunk 2]
```diff
@@ -49,10 +49,10 @@ impl ChunkPoolHandler {
         // When failed to write chunks or finalize transaction in rare case,
         // client need to upload the whole file again.
         while let Some(segment) = segments.pop_front() {
-            self.log_store.put_chunks(file.tx_seq, segment)?;
+            self.log_store.put_chunks(file.tx_seq, segment).await?;
```
Comment (ionian-dev) [inline node/chunk_pool/src/handler.rs:52]: Known issue: This will block processing other messages from `self.receiver`.

**Review by ionian-dev — APPROVED** [review]

File: node/chunk_pool/src/handler.rs (hunk 1 above)
Comment (qhz-ty) [reply node/chunk_pool/src/handler.rs:13]: Just to reuse the async store instance in rpc layer and chunk pool.

File: node/chunk_pool/src/handler.rs (hunk 2 above)
Comment (qhz-ty) [reply node/chunk_pool/src/handler.rs:52]: Current design is to write chunks or finalize tx in single thread, so I used `mpsc::channel` as queue.

File: node/chunk_pool/src/handler.rs (hunk 1 above)
Comment (qhz-ty) [reply node/chunk_pool/src/handler.rs]: Removed `Arc` for async store.

**Review by ionian-dev — APPROVED** [review]

**Pull request merged**